src/preprocess.rkt

```diff
@@ -108,18 +125,31 @@
        (error 'instruction->operator
               "component should always be a subsequence of variables"))
      (define indices (indexes-where variables (curryr member component)))
-     (lambda (points)
-       (let* ([subsequence (map (curry list-ref points) indices)]
+     (lambda (x y)
+       (let* ([subsequence (map (curry list-ref x) indices)]
               [sorted (sort* subsequence)])
-         (list-set* points indices sorted)))]
+         (values (list-set* x indices sorted) y)))]
     [(list 'abs variable)
      (define index (index-of variables variable))
      (define abs (operator-info
                   (get-parametric-operator
                    'fabs
                    (list-ref (context-var-reprs context) index))
                   'fl))
-     (curryr list-update index abs)]))
+     (lambda (x y) (values (list-update x index abs) y))]
+    [(list 'negabs variable)
+     (define index (index-of variables variable))
+     (define negate-variable
+       (operator-info (get-parametric-operator 'neg (list-ref (context-var-reprs context) index)) 'fl))
```

Comment:
Interesting, so you actually use the correct `'neg` over here but the wrong one earlier.